chia/cmds/plotnft.py

```diff
@@ -1,206 +1,255 @@
 from __future__ import annotations
 
-from typing import Optional
+from dataclasses import field
+from pathlib import Path
+from typing import Any, Optional
 
 import click
 
-from chia.cmds import options
-from chia.cmds.param_types import AddressParamType, Bytes32ParamType, CliAddress
+from chia.cmds.cmd_classes import NeedsWalletRPC, chia_command, option
+from chia.cmds.param_types import (
+    AddressParamType,
+    Bytes32ParamType,
+    CliAddress,
+    TransactionFeeParamType,
+)
 from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.errors import CliRpcConnectionError
 from chia.util.ints import uint64
 
 
 @click.group("plotnft", help="Manage your plot NFTs")
-def plotnft_cmd() -> None:
+@click.pass_context
+def plotnft_cmd(ctx: click.Context) -> None:
     pass
 
 
-@plotnft_cmd.command("show", help="Show plotnft information")
-@click.option(
-    "-wp",
-    "--wallet-rpc-port",
-    help="Set the port where the Wallet is hosting the RPC interface. See the rpc_port under wallet in config.yaml",
-    type=int,
-    default=None,
+@chia_command(
+    plotnft_cmd,
+    "show",
+    "Show plotnft information",
 )
-@click.option("-i", "--id", help="ID of the wallet to use", type=int, default=None, show_default=True, required=False)
-@options.create_fingerprint()
-def show_cmd(wallet_rpc_port: Optional[int], fingerprint: int, id: int) -> None:
-    import asyncio
+class ShowPlotNFTCMD:
+    rpc_info: NeedsWalletRPC  # provides wallet-rpc-port and fingerprint options
+    id: Optional[int] = option(
+        "-i", "--id", help="ID of the wallet to use", default=None, show_default=True, required=False
+    )
 
-    from .plotnft_funcs import show
+    async def run(self) -> None:
+        from .plotnft_funcs import show
 
-    asyncio.run(show(wallet_rpc_port, fingerprint, id))
+        async with self.rpc_info.wallet_rpc() as wallet_info:
+            await show(
+                wallet_info=wallet_info,
+                root_path=self.rpc_info.context.get("root_path"),
+                farmer_rpc_port=self.rpc_info.context.get("farmer_rpc_port"),
```

Comment:
used here in the test code https://github.com/Chia-Network/chia-blockchain/blob/21c1ee6f63baf1edf449673ce4adb6e7b6a6e790/chia/_tests/pools/test_pool_cmdline.py#L381

Reply:
So only that test ever uses it , otherwise it's always None

Reply:
`*sigh*` Foiled by leaving my search window filtering out tests...  Hum, I would have guessed we would set the port in the config file in the test to distribute that value.  Then I think this could just not add anything around the rpc port.

Reply:
The port is not set in the config - I think when we changed the tests to use the dynamic port we didn't set ports in the config and instead expect to pass things around. The fixtures all hide this more or less by providing rpc clients already - here I needed to test this code that gets an rpc client.
I could twiddle the config to contain the port - that would might be another option.

Reply:
> I could twiddle the config to contain the port - that would might be another option.

That's what I failed to express.  `:]`  But, if you'd rather just leave a comment about this here instead, I'm fine.  I'm not confident I know the best pattern and as long as it is clear what is going on when we read this I'm ok with it.  Kinda.  Though I think I do kind of prefer the config path, at least a little.  Take your pick.

Reply:
Or maybe we should grow an override CLI option for this port anyways...

Reply:
I switched it to the config - which does seem a little cleaner. Yea, a CLI option might not be a bad idea for a future PR - except `plotnft` is a little special in that is uses the wallet and the farmer - so two ports